### PR TITLE
Add class to manage running a GPSR training and loading the results

### PR DESCRIPTION
**Pull request by lboult**
Hey,

Here's a class that tries to roll together a lot of the workflows for setting up the GPSR training and also allowing models to be loaded in from checkpoints.

I separated it into a lot of functions such that users can override specific bits of it when they need to change something slightly...

Let me know what you think :)

### CODE REVIEW

**Review by roussel-ryan — COMMENTED** [review]
Good start, we want to minimize the number of times the user has to make a subclass (ie. its not required to use this class for a single case). I also think it would be useful to include convenience methods that plot the loss curves or get the most recent version / plot the most recent loss curve

File: gpsr/run.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+            log_name (str): Name for the log directory.
+        """
+        self.hparams = hparams
+        self.log_name = log_name
+        self.train_dataset = None
+
+        print("Hyperparameters:")
+        pprint(self.hparams)
+
+    def setup_training(self, train_dataset):
```
[the remark this answers is not in the document]
Comment (lboult) [reply gpsr/run.py:39]: The reason I had this here is that one also uses the class for loading in already trained models... It could perhaps still be an argument in the initialisation but by default be ```None``` in that case

[the remark this answers is not in the document]
Comment (lboult) [reply gpsr/run.py]: I did this in [e0dc0ea](https://github.com/roussel-ryan/gpsr/pull/43/commits/e0dc0eac1ab7c0ac7e1be0a0fe126b9422a63dcd): the user can still leave the ```train_dataset``` as ```None``` on init though, in the case when they are just loading in a checkpoint rather than training
